src/PrestaShopBundle/Form/Admin/Sell/Product/Pricing/SpecificPriceType.php

```diff
@@ -0,0 +1,306 @@
+     * @param ConfigurableFormChoiceProviderInterface $configurableFormChoiceProvider
+     * @param UrlGeneratorInterface $urlGenerator
+     */
+    public function __construct(
+        TranslatorInterface $translator,
+        array $locales,
+        string $defaultCurrencyIso,
+        FormChoiceProviderInterface $currencyByIdChoiceProvider,
+        FormChoiceProviderInterface $countryByIdChoiceProvider,
+        FormChoiceProviderInterface $groupByIdChoiceProvider,
+        FormChoiceProviderInterface $shopByIdChoiceProvider,
+        FormChoiceProviderInterface $taxInclusionChoiceProvider,
+        ConfigurableFormChoiceProviderInterface $configurableFormChoiceProvider,
+        UrlGeneratorInterface $urlGenerator
+    ) {
+        parent::__construct($translator, $locales);
+        $this->currencyByIdChoiceProvider = $currencyByIdChoiceProvider;
+        $this->countryByIdChoiceProvider = $countryByIdChoiceProvider;
+        $this->groupByIdChoiceProvider = $groupByIdChoiceProvider;
+        $this->shopByIdChoiceProvider = $shopByIdChoiceProvider;
+        $this->taxInclusionChoiceProvider = $taxInclusionChoiceProvider;
+        $this->defaultCurrencyIso = $defaultCurrencyIso;
+        $this->combinationIdChoiceProvider = $configurableFormChoiceProvider;
+        $this->urlGenerator = $urlGenerator;
+    }
+
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        if (!isset($builder->getData()['product_id'])) {
+            // product_id is required for create action and to load combinations choices list, but it is not editable
+            throw new SpecificPriceException('product_id is required to add/edit specific price.');
+        }
+
+        $builder
+            ->add('product_id', HiddenType::class)
+            ->add('currency_id', ChoiceType::class, [
+                'label' => $this->trans('Currency', 'Admin.Global'),
+                'required' => false,
+                'placeholder' => false,
+                'choices' => $this->getModifiedCurrencyChoices(),
+            ])
+            ->add('country_id', ChoiceType::class, [
+                'label' => $this->trans('Country', 'Admin.Global'),
+                'required' => false,
+                'placeholder' => false,
+                'choices' => $this->getModifiedCountryChoices(),
+            ])
+            ->add('group_id', ChoiceType::class, [
+                'label' => $this->trans('Group', 'Admin.Global'),
+                'required' => false,
+                'placeholder' => false,
+                'choices' => $this->getModifiedGroupChoices(),
+            ])
+            ->add('customer', EntitySearchInputType::class, [
+                'label' => $this->trans('Customer', 'Admin.Global'),
+                'layout' => EntitySearchInputType::LIST_LAYOUT,
+                'entry_type' => SearchedCustomerType::class,
+                'remove_modal' => null,
```

Comment:
It seems to me that I misunderstood this change, at first I thought this was done to avoid the modal configuration and rely on default options, or disable the delete feature altogether

But I believe you told me this actually removed the confirmation modal but the delete feature is still usable Seems like it changes the conventional behaviour of all the other lists in the page (or even the BO), a confirmation for deletion is always mandatory AFAIK

Did you change this based on the legacy page? Or is it in the specs somewhere that the modal muste be removed?